design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,180 @@
+
+| **Name**   | **Link**                                                     |
+| ---------- | ------------------------------------------------------------ |
+| JWT Claims | https://auth0.com/docs/tokens/json-web-tokens/json-web-token-claims |
+
+## Requirments
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+1. Required fields
+   1. The user will have the ability to declare in the authenticator policy which claims must be in host annotations. And therefore also in the JWT token.
+   2. Upon authentication request the authenticator MUST validate those claims - existence and content 
+2. Mapping 
+   1. The user should have the ability to map a given "technical" claim name to a more familiar / business one - i.e, in git-lab - instead of mention in the policy "sub" claim, he will have the ability to mention "job"
+
+## Solution
+[//]: # "Elaborate on the solution you are suggesting in this page. Address the functional requirements and the non functional requirements that this solution is addressing. If there are a few options considered for the solution, mention them and explain why the actual solution was chosen over them. Add an execution plan when relevant. It doesn't have to be a full breakdown of the feature, but just a recommendation to how the solution should be approached."
+
+User with define the token schema variables in authenticator policy. They will be checked in the JWT authentication request.
+
+### User Interface
+
+[//]: # "Describe user interface (including command structure, inputs/outputs, etc where relevant)"
+
+#### Authenticator Policy
+
+Two new variables will be added to the authenticator policy:
+
+* mandatory_claims
+* claims_mapping
+
+```yaml
+    - !policy
+    id: conjur/authn-jwt/VendorY
+    body:
+    - !webservice
+    
+    - !variable
+      id: mandatory_claims
+      
+    - !variable
+      id: claims_mapping
+```
+
+#### Variable Values Example
+
+| Name             | Value                   | Description                                      |
+| ---------------- | ----------------------- | ------------------------------------------------ |
+| mandatory_claims | branch, job, aud        | List of claims as they appear in host annotation |
+| claims_mapping   | {ref: branch, sub: job} | Mapping between host annotation to JWT claim     |
+
+For any claim being mapped we will add the host annotation to the `mandatory_claims` variable and not the JWT claim name.
+
+
+## Design
+[//]: # "Add any diagrams, charts and explanations about the design aspect of the solution. Elaborate also about the expected user experience for the feature"
+
+The following class diagram, represent the flow of checking the mandatory claims and the claim mapping.
+
+### Class Diagrams
+[//]: # "Describe classes that are going to be added /changes and their immediate environment. Non-changed classes may be colored differently"
+
+![image-20210711122753500](token-schema-class-diagram.png)
```

Comment:
personal preference,
Make the diagram's boxes closer together and the lettering bigger, otherwise even if we expand it, it will be hard to read. See Oren's or my designs for example :)

Reply:
Great point. Fixed it